### PR TITLE
Throw on ambiguous enums

### LINKED ISSUE #1805: We should fail on ambiguous enum constants as protoc do.

### PR DESCRIPTION
**Pull request by oldergod**
fix #1805

### CODE REVIEW

File: wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/EnumType.kt [hunk 1]
```diff
@@ -59,9 +60,30 @@ data class EnumType(
     if ("true" != allowAlias) {
       validateTagUniqueness(linker)
     }
+    validateTagNameAmbiguity("true" == allowAlias, linker)
     syntaxRules.validateEnumConstants(constants, linker.errors)
   }
 
+  private fun validateTagNameAmbiguity(allowAlias: Boolean, linker: Linker) {
+    val nameToConstants: Map<String, List<EnumConstant>> =
+        constants.groupBy { it.name.toEnglishLowerCase() }
```
Comment (swankjesse) [inline wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/EnumType.kt:69]: What the heck is English? I think somebody means ASCII? English is definitely not stable or formally specified.

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/SchemaTest.kt [hunk 2]
```diff
@@ -2433,4 +2433,65 @@ class SchemaTest {
...
+            |  ZERO = 0;
+            |  zero = 1;
+            |}
+          """.trimMargin())
+          .schema()
+      fail()
+    } catch (e: SchemaException) {
+      assertThat(e).hasMessage(
+          """|Ambiguous constant names (if you are using allow_alias, use the same value for these constants):
+             |  1. ZERO (/source/message.proto:2:3)
```
Comment (swankjesse) [inline wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/SchemaTest.kt:2451]: The 1. and 2. bullets in these is confusing and unnecessary? Maybe use the tag values instead?

**Review by swankjesse — APPROVED** [review]
Great tests.

File: wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/EnumType.kt (hunk 1 above)
Comment (oldergod) [reply wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/EnumType.kt:69]: it uses `toLowerCase(Locale.US)` it's something we're already using in a few places.
I can do manual conversion here instead

File: wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/SchemaTest.kt (hunk 2 above)
Comment (oldergod) [reply wire-library/wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/SchemaTest.kt]:
```
Ambiguous constant names (if you are using allow_alias, use the same value for these constants):
  ZERO:0 (/source/message.proto:3:3)
  zero:1 (/source/message.proto:4:3)
  for enum Foo (/source/message.proto:1:1)
```

**Review by swankjesse — APPROVED** [review]

**Pull request merged**